### PR TITLE
Expose command_name and service_id, do not fail if command respose has empty body

### PR DESCRIPTION
**Pull request by ewertons**

### CODE REVIEW

File: sdk/inc/azure/core/az_mqtt5_rpc_server.h [hunk 1]
```diff
@@ -164,6 +164,14 @@ typedef struct
...
    */
   az_span content_type;
+  /**
+   * @brief The service id of the request.
+   */
+  az_span service_id;
+  /**
+   * @brief The command name of the request.
+   */
+  az_span command_name;
```
Comment (vaavva) [inline sdk/inc/azure/core/az_mqtt5_rpc_server.h:174]: @RLeclair was there a reason we didn't include the command name in this before?

File: sdk/src/azure/core/az_mqtt5_rpc_server_hfsm.c [hunk 2]
```diff
@@ -244,6 +244,8 @@ AZ_INLINE az_result _handle_request(az_mqtt5_rpc_server* this_policy, az_mqtt5_r
             .request_data = data->payload,
             .request_topic = data->topic,
             .content_type = content_type_str,
+            .command_name = req_event_data->command_name,
```
Comment (vaavva) [inline sdk/src/azure/core/az_mqtt5_rpc_server_hfsm.c:247]: Does this need to be copied/will there be any issues with the span getting deallocated?

**Review by vaavva — APPROVED** [review]

File: sdk/src/azure/core/az_mqtt5_rpc_server_hfsm.c (hunk 2 above)
Comment (ewertons) [reply sdk/src/azure/core/az_mqtt5_rpc_server_hfsm.c:247]: These data are only valid in the scope of the callback. The target of the callback must copy the information if they need it. After cb and _handle_request return this data is disregarded.

File: sdk/inc/azure/core/az_mqtt5_rpc_server.h (hunk 1 above)
Comment (RLeclair) [reply sdk/inc/azure/core/az_mqtt5_rpc_server.h:174]: This should have been included here, the change is correct

**Review by RLeclair — APPROVED** [review]

**Comment by ewertons** [comment]
Found the issue with test failures. Missing test was not updated to expect empty payloads in command responses. 

I modified and renamed test_az_mqtt5_rpc_client_recv_response_no_payload_failure to test_az_mqtt5_rpc_client_recv_response_no_payload_success
